content/en/docs/advanced-tasks/fail-forward-upgrades.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Fail forward upgrades"
+linkTitle: "Fail forward upgrades"
+weight: 4
+description: >
+    Fail forward upgrades
+---
+
+## What "Fail Forward" upgrades add
+A failed installation/upgrade is typically caused by one of two scenarios:
+1. **A Failed CSV:** The CSV is in the FAILED phase.
+2. **A Failed InstallPlan:** Usually occurs because a resource listed in the InstallPlan fails to be created or updated. An InstallPlan may fail independently of its CSV and may fail to create the CSV.
+
+By opting into "failed forward" upgrades, OLM will allow you to recover from failed installations and upgrades by:
```

Comment:
We mention opting into this functionality but don't document until header closer to the end of this documentation. Should we at least link to that header, or do we need to restructure this?